test/validate/pass_pointer_exists.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/sh
+
+set -o errexit
+set -o nounset
+
+TMP="$(mktemp -d)"
+cleanup() { rm -rf "$TMP"; }
+trap cleanup EXIT
+
+cat << 'EOF' > "$TMP/schema.json"
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "nested": {
+      "type": "object",
+      "properties": {
+        "foo": { "type": "integer" }
+      },
+      "required": ["foo"]
+    }
+  }
+}
+EOF
+
+cat << 'EOF' > "$TMP/instance.json"
+{
+  "foo": 123
+}
+EOF
+
+if ! "$1" validate \
```

Comment:
Also, you don't need the explicit `if` check. Because of `set -o errexit`, the entire script will fail if any of its commands fail. See how we do it for other "pass" tests